src/components/sbb-toggle/sbb-toggle.tsx

```diff
@@ -0,0 +1,202 @@
+ */
+
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-toggle.scss',
+  tag: 'sbb-toggle',
+})
+export class SbbToggle implements ComponentInterface {
+  /**
+   * Id of the toggle element.
+   */
+  @Prop() public sbbToggleId = `sbb-toggle-${++nextId}`;
+
+  /**
+   * Whether the toggle is disabled.
+   */
+  @Prop({ reflect: true }) public disabled = false;
+
+  /**
+   * If true set the width of the component fixed; if false the width is dynamic based on the label of the sbb-toggle-option.
+   */
+  @Prop({ reflect: true }) public even: boolean;
+
+  /**
+   * Size variant, either m or s.
+   */
+  @Prop({ reflect: true }) public size?: InterfaceSbbToggleAttributes['size'] = 'm';
+
+  /**
+   * The value of the toggle.
+   */
+  @Prop({ mutable: true }) public value: any | null;
+
+  /**
+   * Whether the animation is enabled.
+   */
+  @Prop({ reflect: true, mutable: true }) public disableAnimation = false;
+
+  @Element() private _element: HTMLElement;
+
+  private _toggleElement: HTMLElement;
+  private _toggleResizeObserver = new ResizeObserver(() => this._setCheckedPillPosition(true));
+
+  @Watch('value')
+  public valueChanged(value: any | undefined): void {
+    for (const toggleOption of this._options) {
+      toggleOption.checked = toggleOption.value === value;
+      toggleOption.tabIndex = this._getOptionTabIndex(toggleOption);
+    }
+    this._setCheckedPillPosition(false);
+    this.change.emit({ value });
+    this.didChange.emit({ value });
+  }
+
+  @Watch('disabled')
+  public updateDisabled(): void {
+    for (const toggleOption of this._options) {
+      toggleOption.disabled = toggleOption.disabled ? toggleOption.disabled : this.disabled;
+      toggleOption.tabIndex = this._getOptionTabIndex(toggleOption);
+    }
+  }
+
+  /**
+   * Emits whenever the radio group value changes.
+   * @deprecated only used for React. Will probably be removed once React 19 is available.
+   */
+  @Event({
+    bubbles: true,
+    composed: true,
+  })
+  public didChange: EventEmitter;
+
+  /**
+   * Emits whenever the radio group value changes.
+   */
+  @Event({
+    bubbles: true,
+    composed: true,
+  })
+  public change: EventEmitter;
+
+  private get _options(): HTMLSbbToggleOptionElement[] {
+    return Array.from(
+      this._element.querySelectorAll('sbb-toggle-option')
+    ) as HTMLSbbToggleOptionElement[];
+  }
+
+  private get _checked(): HTMLSbbToggleOptionElement {
+    return this._options.find((toggle) => toggle.checked) ?? this._options[0];
+  }
+
+  @Listen('did-select', { passive: true })
+  public onToggleOptionSelect(event: CustomEvent): void {
+    this.value = event.detail;
+  }
+
+  private _setCheckedPillPosition(disableAnimation: boolean): void {
+    const options = this._options;
+    const checked = this._checked;
+
+    if (!checked) {
+      return;
+    }
+
+    if (!disableAnimation) {
+      this._element.removeAttribute('disable-animation');
```

Comment:
Is this part really needed?

Reply:
This is needed otherwise the first click will not be animated.